activemq-broker/src/main/java/org/apache/activemq/broker/scheduler/SchedulerBroker.java

```diff
@@ -353,6 +355,12 @@ private void doSchedule(Message messageSend, Object cronValue, Object periodValu
 
         getInternalScheduler().schedule(jobId.toString(),
                 new ByteSequence(packet.data, packet.offset, packet.length), cronEntry, delay, period, repeat);
+
```

Comment:
@mattrpav this block is the main reason for a PR. As there is no way to get to the job id at the time of the scheduling.